Packs/EmailCommunication/Scripts/PreprocessEmail/PreprocessEmail.py

```diff
@@ -90,19 +83,25 @@ def main():
     email_subject = custom_fields.get('emailsubject')
 
     try:
+
         email_related_incident = int(email_subject.split('#')[1].split()[0])
         query = f"id:{email_related_incident}"
         incident_details = get_incident_by_query(query)
         check_incident_status(incident_details, str(email_related_incident))
         email_reply = set_email_reply(email_from, email_to, email_cc, email_body)
-
         add_entries(email_reply, str(email_related_incident))
+
+        # False - to not create new incident
         demisto.results(False)
 
-    except (IndexError, ValueError):
-        demisto.results("exit value")
+    except (IndexError, ValueError) as e:
+        demisto.results(True)
+        return_error(f"PreprocessEmail script has failed - {e}")
+
+    #
+    except DemistoException as e:
         demisto.results(True)
-        sys.exit(0)
+        return_error(f"PreprocessEmail script has failed - {e}")
```

Comment:
Im not sure that this error is true here, aren't we use the `DemistoException` to indicate the need for a new incident?

Reply:
done